packages/ve-hemi-rewards/actions/wallet/collectRewards.ts

```diff
@@ -0,0 +1,143 @@
+import { EventEmitter } from 'events'
+import type { Address, TransactionReceipt, WalletClient } from 'viem'
+import { encodeFunctionData, zeroAddress } from 'viem'
+import { waitForTransactionReceipt, writeContract } from 'viem/actions'
+
+import { veHemiRewardsAbi } from '../../abi'
+import { getVeHemiRewardsContractAddress } from '../../constants'
+import type { CollectAllRewardsEvents } from '../../types'
+import { toPromiseEvent } from '../../utils'
+
+const canCollectRewards = async function ({
+  rewardToken,
+  tokenId,
+  walletClient,
+}: {
+  rewardToken: Address
+  tokenId: bigint
+  walletClient: WalletClient
+}): Promise<{
+  canCollect: boolean
+  reason?: string
+}> {
+  if (!walletClient.chain) {
+    return {
+      canCollect: false,
+      reason: 'wallet client chain is not defined',
+    }
+  }
+
+  if (!tokenId || tokenId <= BigInt(0)) {
+    return {
+      canCollect: false,
+      reason: 'invalid token id',
+    }
+  }
+
+  if (!rewardToken || rewardToken === zeroAddress) {
```

Comment:
I think here we should also check for `isAddress`. Let's remember types are not enforced on runtime